extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSourceFactory.java

```diff
@@ -86,17 +86,27 @@ public DataSource createSource(DataFlowRequest request) {
         var queryParams = request.getProperties().get(QUERY_PARAMS);
 
         try {
-            return Result.success(HttpDataSource.Builder.newInstance()
+            var httpDataSourceBuilder = HttpDataSource.Builder.newInstance()
                     .httpClient(httpClient)
                     .sourceUrl(endpoint)
                     .name(name)
-                    .method(method)
-                    .header(authKey, authCode)
-                    .queryParams(queryParams)
+                    .method(method);
+
+            if (authKey != null && authCode != null) {
+                httpDataSourceBuilder.header(authKey, authCode);
+            }
+
+            if (queryParams != null) {
+                httpDataSourceBuilder.queryParams(queryParams);
+            }
+
+            httpDataSourceBuilder
                     .requestId(request.getId())
                     .retryPolicy(retryPolicy)
-                    .monitor(monitor)
-                    .build());
+                    .monitor(monitor);
```

Comment:
nit: move up so all non-optional builder calls are done in one train?

Reply:
Update it.